src/scene/shader-lib/programs/shader-generator.js

```diff
@@ -13,6 +14,16 @@ class ShaderGenerator {
         return '}\n';
     }
 
+    /**
+     * @param {Set<string>} defines - the set of defines to be used in the shader.
+     * @returns {number} the hash code of the defines.
+     */
+    static definesHash(defines) {
+        const keys = Array.from(defines.keys()).sort();
```

Comment:
does this need updating to support define values?

Reply:
yep, most definitely, done and thanks!